app/models/foreman_tasks/concerns/host_action_subject.rb

```diff
@@ -15,26 +15,30 @@ def available_locks
       module ClassMethods
         # TODO: This should get into the Foreman core, extracting the
         # +importHostAndFacts+ method into two
-        def importHost(hostname, certname, proxy_id = nil)
+        def import_host(hostname, certname, facts, proxy_id = nil)
+          raise(::Foreman::Exception.new("Invalid Facts, must be a Hash")) unless facts.is_a?(Hash)
           raise(::Foreman::Exception.new("Invalid Hostname, must be a String")) unless hostname.is_a?(String)
 
           # downcase everything
           hostname.try(:downcase!)
           certname.try(:downcase!)
+          facts['domain'].try(:downcase!)
```

Comment:
nowhere, probably should be moved to facts importing when https://github.com/theforeman/foreman-tasks/pull/176/files#diff-a2cb9fe73d15f4e6310e2ef57f76e22bR16 is implemented, good catch but I'm not sure where to put it here, I don't want to wrap another method (Host::Base#import_facts)... I plan to open the split PR against core soon